### DISCUSSION TITLE
Add Delete Rule to GUI when setting Relationships

### DISCUSSION THREAD

**Discussion by Lior539** [supabase/supabase#12329] [Feature Requests]
# Feature request

Right now, when using the GUI to add a relationship between tables you cannot set the delete rule (e.g. cascade). You almost always want to specify the delete rule on the relationship when creating it, but this limitation in the GUI means that effectively the GUI has no use when it comes to setting relationship and you always need to use the SQL editor instead.

See attached screenshot of current GUI:
<img width="459" alt="Screenshot_2022-06-15_at_11 19 19" src="https://user-images.githubusercontent.com/7090054/173809778-c7e2928e-7489-44b8-bd75-6d0cbbea248a.png">

**Comment by sannajammeh** [comment]
+1

**Comment by ryanrussell** [comment]
@Lior539

Echoing this sentiment on **referential integrity around cascading deletes**, having to go to the editor and do an `ALTER TABLE ... DROP CONSTRAINT ...  ADD CONSTRAINT` to add an `on delete cascade` seems like it could be made into a checkbox in the GUI.

Please and thank you :)

**Comment by cullophid** [comment]
👍 this would be fantastic. Right now I had to set up all the relationships in SQL.
